design-documents/mde-project-structure.md

```diff
@@ -0,0 +1,41 @@
+# Overview
+
+There is a need to have unified approach to structuring the Git-based project while developing MDEs (Magento Developed Extensions), like Page Builder and MSI.
+
+Currently, MSI is developed in a fork of main Magento repository, while Page Builder is developed in a separate repository (like B2B).
+
+## MDE project structure for Git-based installations
+
+MDE project must be structured as Magento B2B:
+ 1. Modules must be placed under `app/code`
```

Comment:
Is it only because of non-modular tests?
Based on the previous agreement, the directory structure should be:
```
<repo_root>/
           <Module1>
                Test/
                    Unit/
                    Integration/
                    Mftf/
                               TestSuite/
                               composer.json
           <Module2>
           <Module1SampleData>
           <Module2SampleData>
```

So there should be no `app/code`.

See https://wiki.corp.magento.com/display/MAGE2/Bundled+Extensions+Repository+Structure+--+HLD (internal link).

Reply:
I would say this is because of EE linking tool

Reply:
I agree with @buskamuza and we already made a lot of effort to adopt the described structure (without app/code folder) or prepare source code for simple migration in future. As an example, all MSI Inventory modules contains integration tests inside Test/Integration modules. This is not resolve issue with integration tests modularity but make it more simple in future.

Reply:
I like idea to have modules in the root of repository. With this and meaningful module names visitor will get understanding what this project is about and what features provide. It should not be complex or lead to bad performance to adapt tool for such structure